services/server/src/lib/fawry.ts

```diff
@@ -0,0 +1,47 @@
+import { GeneralRequestPayload } from "@/fawry/types/requests";
+import { PaymentMethod } from "@/fawry/types/ancillaries";
+import { fawryConfig } from "@/constants";
+import { IUser } from "@litespace/types";
+
+export function forgeFawryPayload<Other = object>({
+  merchantRefNum,
+  amount,
+  paymentMethod,
+  signature,
+  user,
+  chargeItems,
+  description,
+  orderWebHookUrl,
+  other,
+}: {
+  merchantRefNum: number;
+  amount: number;
+  paymentMethod: PaymentMethod;
+  signature?: string;
+  user?: IUser.Self;
+  chargeItems?: GeneralRequestPayload["chargeItems"];
+  description?: string;
+  orderWebHookUrl?: string;
+  other?: Other;
+}) {
+  return {
+    merchantRefNum,
+    paymentMethod,
+    amount,
+    signature: signature || "",
+
+    customerProfileId: user?.id,
+    customerName: user?.name,
+    customerMobile: user?.phone,
+    customerEmail: user?.email,
```

Comment:
The phone and the email is required as estated in the fawry docs. You are don't getting any errors about it because you are using casting.